Enable media element for castanets

sm-venugopal opened the pull request:
1. Initial patch supporting media element playback
2. Added video hole support for video playback
3. Added video hole preference to WebView
4. Disabled android specific features

joonghunpark commented:
Which device configuration scenarios does this CL aiming at?
Tizen(browser process) and Tizen, Android(renderer process)?

joonghunpark reviewed (commented):
LGTM. But please wait for review from someone others.

sm-venugopal commented:
> Which device configuration scenarios does this CL aiming at?
> Tizen(browser process) and Tizen, Android(renderer process)?

Thats right.

joonghunpark commented on content/renderer/media/castanets/castanets_webmediaplayer_impl.cc:
```diff
@@ -0,0 +1,744 @@
+  return 0;
+};
+
+bool WebMediaPlayerCastanets::CopyVideoTextureToPlatformTexture(
```
This method only returns false.
If this method will have more implementation,
please leave a TODO(). 
Otherwise either 'early return false' or 'NOTIMPLEMENTED();' would be better, I think.

Chikdolman commented:
Requested to rebuild the Android, as the android build fail occurred due to the network problem from the build server.

sm-venugopal committed:
Disable android specific features

Target running browser process need not support these features.
These may cause unpredicted behaviour.

Signed-off-by: Venugopal S M <sm.venugopal@samsung.com>

sm-venugopal committed:
Set video hole preference to WebView

Added video hole preference setter/ getter. Video hole preference
is also applied to each WebView.

Signed-off-by: Venugopal S M <sm.venugopal@samsung.com>

sm-venugopal committed:
Adding video hole support

A transparent quad is drawn to show through the video played
at HW plane.

This commit introduces VIDEO_HOLE flag.

Signed-off-by: Venugopal S M <sm.venugopal@samsung.com>

sm-venugopal committed:
Alter media player interface and type enum

- Added video hole types to media player type class. The order should be
  same as castanet player.
- New APIs are added to interface class.

Signed-off-by: Venugopal S M <sm.venugopal@samsung.com>

sm-venugopal committed:
Initial patch supporting media element playback

- Added interface for castanets WebMediaPlayer
- Added renderer manager for managing WebMediaPlayer objects
- Added IPCs

Signed-off-by: Venugopal S M <sm.venugopal@samsung.com>

sm-venugopal commented:
Updated with Video hole support for inline playback.
